lucuma-schemas/src/test/scala/lucuma/schemas/encoders/PosAngleEncodersSuite.scala

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.schemas.encoders
+
+import cats.effect.IO
+import io.circe.Encoder
+import lucuma.schemas.decoders._
+import lucuma.core.model.{ PosAngle => PosAngleConstraint }
+
+class PosAngleEncodersSuite extends InputStreamSuite {
```

Comment:
Aren't there circe law testing methods?

Reply:
Yes, we weren't really set up for it which surprised me.  I thought there must be a good reason, but maybe it's just that we had no encoders before.  At any rate, I added a `PosAngleConstraintCodecSuite` with

```
  checkAll("PosAngleConstraint Codec", CodecTests[PosAngleConstraint].codec)
```

Reply:
Exactly, we haven't encountered a situation yet where we have to share encoders for regular types among projects. 

The idea was that that this library be shared among client code. They do need the encoders for the input types, which are generated by clue here.

If this library is to be used by the ODB API it creates a soft cyclic dependency, since the ODB schema is technically an input here.

In the future, I was thinking maybe we could dissolve this project and split it into subprojects of the server implementations (ODB, ITC, etc.), each spinning up the local server and reading the schema from there instead of us updating it manually. That way, each server would also provide its own client library.

Reply:
If we're updating a `PosAngleConstraint`, we won't be encoding directly. Don't all of the mutations require a `PosAngleConstraintInput`? Would we ever be sending a `PosAngleConstraint` to the API?

Reply:
Thanks for clarifying the usage. The API then has an input type for PosAngleConstraint?

What we have been doing in those cases is use the generated input type for communicating with the ODB and implement conversion from our model to the input type like we do here: https://github.com/gemini-hlsw/explore/blob/master/common/src/main/scala/explore/schemas/odb/implicits.scala#L24.

We should probably move those conversions over to this library.

Reply:
> What we have been doing in those cases is use the generated input type for communicating with the ODB and implement conversion from our model to the input type like we do here

Ah ha, I see.  Yes then the encoder won't be used but we'll need something like what it does in one of these `toInput` ops.  We can delete the encoder I think.

Reply:
I could do that, since I'm the one who started all of this...

Reply:
Though, a complication here is that the API represents the core `Unconstrained` object as the lack of a constraint.  So there is no corresponding `PosAngleConstraintInput` and instead you can delete any existing constraint by assigning it `null`.  In other words, given `Unconstrained` there is no `PosAngleConstraintInput` that matches it.

I suppose we can make `Unconstrained` a value instead of the lack of a value but an "unconstrained" constraint seems wrong to me.

Reply:
In some cases we have the `toInput` methods return an `Input[A]`. In this case, it could translate `Unscontrained` to `Unassign` and the other cases to `Assign(translation)`.

Reply:
Sounds perfect.